planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_planner_module.cpp

```diff
@@ -923,54 +930,44 @@ std::vector<PullOverPath> GoalPlannerModule::sortPullOverPathCandidatesByGoalPri
 
   // if object recognition is enabled, sort by collision check margin
   if (parameters_->use_object_recognition) {
-    // (2) Sort by collision check margins
-    const std::vector<double> margins = std::invoke([&]() {
-      std::vector<double> combined_margins = soft_margins;
-      combined_margins.insert(combined_margins.end(), hard_margins.begin(), hard_margins.end());
-      return combined_margins;
-    });
-
-    // Create a map of PullOverPath pointer to largest collision check margin
-    auto calcLargestMargin = [&](const PullOverPath & pull_over_path) {
-      // check has safe goal
-      const auto goal_candidate_it = std::find_if(
-        goal_candidates.begin(), goal_candidates.end(),
-        [pull_over_path](const auto & goal_candidate) {
-          return goal_candidate.id == pull_over_path.goal_id;
-        });
-      if (goal_candidate_it != goal_candidates.end() && !goal_candidate_it->is_safe) {
-        return 0.0;
-      }
-      // check path collision margin
-      const PathWithLaneId parking_path = pull_over_path.getParkingPath();
-      const std::vector<double> parking_path_curvatures = pull_over_path.getParkingPathCurvatures();
-      for (const auto & margin : margins) {
-        if (!checkObjectsCollision(
-              parking_path, parking_path_curvatures, planner_data_, *parameters_, margin,
-              /*extract_static_objects=*/true)) {
-          return margin;
-        }
-      }
-      return 0.0;
-    };
+    // STEP2-2: Sort by collision check margins
+    const auto [margins, margins_with_zero] =
+      std::invoke([&]() -> std::tuple<std::vector<double>, std::vector<double>> {
+        std::vector<double> margins = soft_margins;
+        margins.insert(margins.end(), hard_margins.begin(), hard_margins.end());
+        std::vector<double> margins_with_zero = margins;
+        margins_with_zero.push_back(0.0);
+        return std::make_tuple(margins, margins_with_zero);
+      });
 
     // Create a map of PullOverPath pointer to largest collision check margin
     std::map<size_t, double> path_id_to_margin_map;
```

Comment:
Can you rename path_id_to_margin_map to path_id_to_rough_margin_map ?

Reply:
that is better!
fixed in https://github.com/autowarefoundation/autoware.universe/pull/8457/commits/9b43e03d75ff09da2f471f98ca5eb305a2817b5e